backend/onedrive/onedrive.go

```diff
@@ -1503,7 +1503,75 @@ func (f *Fs) PublicLink(ctx context.Context, remote string, expire fs.Duration,
 		fmt.Println(err)
 		return "", err
 	}
-	return result.Link.WebURL, nil
+
+	shareURL := result.Link.WebURL
+
+	// Convert share link to direct download link if target is not a folder
+	cnvFailMsg := "Don't know how to convert share link to direct link - returning the link as is"
+	if info.Folder != nil {
+		fs.Logf(f, cnvFailMsg)
```

Comment:
```suggestion
		fs.Debugf(nil, "Can't convert share link for folder to direct link - returning the link as is")
```

IMO we're not expecting share links for folders to be convertible so we don't need to disturb the user